core/src/connection/error.rs

```diff
@@ -84,8 +84,8 @@ pub enum PendingConnectionError<TTransErr> {
     Aborted,
 
     /// The peer identity obtained on the connection did not
-    /// match the one that was expected or is otherwise invalid.
-    InvalidPeerId,
+    /// match the one that was expected or is the local one.
+    InvalidPeerId(PeerId),
```

Comment:
```suggestion
    InvalidPeerId {
      obtained: PeerId,
    }
```

What do you think? That would make it clear that the provided one in `InvalidPeerId` is the one obtained on the connection.

Reply:
Yes, and I’ll want to add the failed address if possible (looking into it now).